concur.go

```diff
@@ -29,15 +31,19 @@ func (s *server0) order(a address) {
 // exercise 1
 type server1 struct {
 	sh shipper
+	mu *sync.Mutex
```

Comment:
You can even just leave it as a sync.Mutex (not a pointer), and then you don't need to initialize it.